packages/wrangler/src/index.ts

```diff
@@ -1172,10 +1172,12 @@ export async function main(argv: string[]): Promise<void> {
 			}
 
 			await closeSentry();
+			const controller = new AbortController();
+
 			await Promise.race([
-				await Promise.allSettled(dispatcher?.requests ?? []),
-				setTimeout(1000), // Ensure we don't hang indefinitely
-			]);
+				Promise.allSettled(dispatcher?.requests ?? []),
+				setTimeout(1000, undefined, controller), // Ensure we don't hang indefinitely
+			]).then(() => controller.abort()); // Ensure the Wrangler process doesn't hang waiting for setTimeout(1000) to complete
```

Comment:
Super Nit: `Promise.allSettled` won't throw, so this shouldn't have any actual difference 🤔 

```suggestion
			]).finally(() => controller.abort()); // Ensure the Wrangler process doesn't hang waiting for setTimeout(1000) to complete
```

Reply:
Not really, just so we can guarantee the controller is aborted if it throws for whatever reasons. But it shouldn't have any actual difference with the current promise setup 🤷